mautic/mautic
# Campaign summary statistics

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | campaign-summary 
| Bug fix?                               | 
| New feature?                           | Yes
| Deprecations?                          | 
| BC breaks?                             | 
| Automated tests included?              | Yes
| Related user documentation PR URL      |  
| Related developer documentation PR URL |  
| Issue(s) addressed                     | [#6651](https://github.com/mautic/mautic/pull/6651) 



**ToDo**

- [x] Write more functional tests to guarantee the charts and counts work as they should.
- [x] Fix the command as it seems it does not count contacts properly. There is a functional test that shows it.
- [x] Implement the Pending Count that has been added after this PR was submitted. It shows bad counts because of it after rebase.


#### Description:
This PR fixes [#6651](https://github.com/mautic/mautic/pull/6651).



Need run migration: `php bin/console doctrine:migrations:execute 20201120122846`

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to /s/config/edit and Campaign Settings.
3. Enable "Use summary statistics" and "Use date range"
4. Progress a contact through any campaign.
5. Go to the campaign /s/campaigns and view the Campaign Statistics chart. It should display the events exactly as you would expect.
6. View the Campaign Actions/Conditions/Decisions tabs (if there are any) and they should show the same results they always have as well.
7. Pick a date range with no contacts and hit apply. You'll see the campaign Actions/Conditions/Decisions/Contacts tabs disappear or contain no content. These tabs now respect the date range provided.



<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9471"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
